spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigClientProperties.java

```diff
@@ -216,9 +216,9 @@ public void setHeaders(Map<String, String> headers) {
 		this.headers = headers;
 	}
 
-	private Credentials extractCredentials() {
+	private Credentials extractCredentials(int index) {
 		Credentials result = new Credentials();
-		String uri = this.uri;
+		String uri = this.uri[index];
```

Comment:
I would make sure this isnt going to throw an `ArrayIndexOutOfBoundsException`

Reply:
I get your point but this is being called from only one place in the code, where we are in a for loop, iterating through an array of urls. The terminating condition for that loop is the length of the array itself, so this exception should never occur. 
Though, I will definitely add checks to be bullet-proof!